python/pyspark/pandas/indexes/base.py

```diff
@@ -2000,11 +2010,16 @@ def difference(self, other: "Index", sort: Optional[bool] = None) -> "Index":
                     sort
                 )
             )
-        # Handling MultiIndex
+        # Handling MultiIndex when `other` is list-like objects.
         if isinstance(self, MultiIndex) and not isinstance(other, MultiIndex):
```

Comment:
can you extract this condition and define it in a variable? It's difficult to follow the if-elses here.